datafusion/sql/src/select.rs

```diff
@@ -70,10 +71,37 @@ impl<'a, S: ContextProvider> SqlToRel<'a, S> {
         // process `where` clause
         let plan = self.plan_selection(select.selection, plan, planner_context)?;
 
+        // handle named windows before processing the projection expression
+        let mut modified_projection = select.projection.clone();
+        // If the projection is done over a named window, that window
+        // name must be defined. Otherwise, it gives an error.
+        for proj in modified_projection.iter_mut() {
+            if let SelectItem::ExprWithAlias {
+                expr: SQLExpr::Function(f),
+                alias: _,
+            } = proj
+            {
+                for NamedWindowDefinition(window_ident, window_spec) in
+                    select.named_window.iter()
+                {
+                    if let Some(WindowType::NamedWindow(ident)) = &f.over {
+                        if ident.eq(window_ident) {
+                            f.over = Some(WindowType::WindowSpec(window_spec.clone()))
+                        }
+                    }
+                }
+                // All named windows must be defined with a WindowSpec.
```

Comment:
Thanks for the review, I agree with you. It would be better to add such a control. I think even if that multiple-defined window is not used, we should give an error, am I right?

Reply:
I recommend we do whatever postgres does (which I assume is throw an error, but I haven't checked)

Reply:
I have checked, it says "Query Error: error: window "window3" is already defined"